service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformUtils.java

```diff
@@ -62,6 +65,45 @@ public static void setCommandError(Command command, Exception e) {
                 .build());
     }
 
+    /**
+     * Delete the Terraform state directory for the given request under the given directory.<br>
+     * The directory name is constructed as follows: {context}-{serviceId}
+     * @param request
+     * @param directory
+     */
+    public static void deleteConfigPath(CommandRequest request, String directory) {
```

Comment:
Although not relevant now, but this method is not thread safe. 
However, I do not think we will see any race condition (because only one config push job for a given application can be in progress and the directory structure is unique for given broker and application)

Reply:
making this method thread safe might prevent future usage - thus added a simple lock object. We should make the other methods thread safe, too, but not now.